src/ship.cpp

```diff
@@ -30,18 +33,52 @@ void Ship::draw(Camera &camera) {
 }
 
 void Ship::move() {
+    /* Difference between Ship speed and Wind speed 
+    proportional to the angle of the ship to the wind 
+    This calculation could change depending type of sails */
+    int w_speed = Weather::wind_speed;
+    radians w_direction = Weather::wind_direction;
+    float push = (w_speed-speed) * (cos(w_direction-rotation)+1)/2;
+    /* Hull drag could be a Ship modifier */
+    float drag = 0.05;
+    /* Sails could be a Ship Modifier 
+    * Change Rate could depend on crew manning sails
+    */
+    float sails_type = 0.8;
+    float sails_change_rate = 0.1;
+    /* Sails calculation */
+    sails_efficiency += (sails_aperture - sails_efficiency)*sails_change_rate;
+    /* New speed with inertia */
+    float acceleration = 0.02;
+    goal_speed = (1-drag)*sails_type*(speed + push*sails_efficiency);
+    speed += (goal_speed-speed)*acceleration;
+
     x -= speed * sin(rotation);
     y -= speed * cos(rotation);
 }
 
 void Ship::turn_left(float rotate) {
```

Comment:
Wouldn't it make more sense if turn_right and turn_left were unified as turn(angle), with the angle being clockwise or something? Otherwise, at least the implementation could be unified as a private turn(angle) and turn_left/right could call it with opposed signs.